shared/src/question-generators/demos/multiInput.ts

```diff
@@ -0,0 +1,81 @@
+import { MultiFreeTextQuestion, QuestionGenerator } from "@shared/api/QuestionGenerator"
+import { serializeGeneratorCall } from "@shared/api/QuestionRouter"
+import Random from "@shared/utils/random"
+import { t, tFunctional, Translations } from "@shared/utils/translations"
+
+/**
+ * All displayed text goes into the translations object.
+ * Include at least english and german.
+ */
+const translations: Translations = {
+  en: {
+    name: "Multi Input Question",
+    description: "Compute the sum of two integers",
+    text: "Solve the following sums: {{test1#NL#A) {{0}} + {{1}} = #sum}} {{test2#NL#B) {{2}} + {{3}} = #sum}}",
+  },
+  de: {
+    name: "Frage mit mehreren Eingabefeldern",
+    description: "Berechne die Summe zweier Zahlen",
+    text: "Löse die folgenden Summen:\n {{test1#x#A) {{0}} + {{1}} = #Summe}}\n {{test2#y#B) {{2}} + {{3}} = #Summe}}",
+  },
+}
+
+/**
+ * This question generator generates a simple multiple choice question.
+ */
+export const DemoMultiInput: QuestionGenerator = {
+  id: "demomi",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["demo"],
+  languages: ["en", "de"],
+  license: "MIT",
+  expectedParameters: [],
+
+  /**
+   * Generates a new MultipleChoiceQuestion question.
+   *
+   * @param lang The language of the question
+   * @param parameters The parameters for the question. In this case none are used.
+   * @param seed The seed for the random number generator
+   * @returns A new MultipleChoiceQuestion question
+   */
+  generate: (lang = "en", parameters, seed) => {
+    // initialize the RNG so the question can be generated again
+    const random = new Random(seed)
+
+    // generate the question values
+    const a = [random.int(5, 15), random.int(5, 15), random.int(5, 15), random.int(5, 15)]
+    const b = [random.int(5, 15), random.int(5, 15), random.int(5, 15), random.int(5, 15)]
+
+    // generate the question object
+    const question: MultiFreeTextQuestion = {
+      type: "MultiFreeTextQuestion",
+      name: DemoMultiInput.name(lang),
+      path: serializeGeneratorCall({
+        generator: DemoMultiInput,
+        lang,
+        parameters,
+        seed,
+      }),
+      fillOutAll: true,
+      text: t(translations, lang, "text", [`${a[0]}`, `${b[0]}`, `${a[1]}`, `${b[1]}`]),
+      feedback: (answer) => {
+        return {
+          correct:
+            parseInt(answer.text["test1"]) === a[0] + b[0] &&
+            parseInt(answer.text["test2"]) === a[1] + b[1],
+          correctAnswer: `${a[0]} + ${b[0]} = ${a[0] + b[0]}, ${a[1]} + ${b[1]} = ${a[1] + b[1]}`,
+        }
+      },
+      checkFormat: ({ text }, fieldID) => {
+        const val = parseInt(text[fieldID])
```

Comment:
They are two separate functions. Not sure what you mean here.

Reply:
Mh I meant a top-level extracting. 
`function checkFormat(): MultiFreeTextFormat... { ... }` and then using this. 
So if someone new is reading this code, I think it would be simpler to understand. 
But its not that important

Reply:
Ah ok. Since the functions are very short I would keep them lined in.

Reply:
Okay, we'll add more demos from time to time, then there will probably be other generators with non inlined check format / feedback 👍